### ISSUE TITLE
problem with arguments idea

### ISSUE THREAD

**Issue by tunnckoCore** [phated/async-done#23]
Problem is that [index.js#L26](https://github.com/phated/async-done/blob/master/index.js#L26) limit us. Currently that limit us to pass only one value to the callback e.g.

```js
asyncDone(function (done) {
  done(null, 123);
}, function(err, res) {
  console.log(res)
  //=> 123
})
```
but you cant 
```js
asyncDone(function (done) {
  done(null, 123, 456);
}, function(err, res) {
  console.log(res)
  //=> 123
})
```

One way to fix it is to be 1:1 relation
```js
asyncDone(function (done) {
  done(null, 123, 456);
}, function(err, foo, bar) {
  console.log(foo) //=> 123
  console.log(bar) //=> 456
})
```
to do it we should change `done fn` to this
```js
function done(){
  d.removeListener('error', onError);
  d.exit();

  return cb.apply(null, arguments);
}
```

Or other way is like `.then` do it - single array with values.
```js
asyncDone(function (done) {
  done(null, 123, 456);
}, function(err, results) {
  console.log(results)
  //=> [123, 456]
})
```

**Comment by tunnckoCore** [comment]
Im more for 1:1

**Comment by phated** [comment]
Published 1:1 in version 1.1.0
